storage/innobase/srv/srv0srv.cc

```diff
@@ -1877,7 +1883,10 @@ DECLARE_THREAD(srv_error_monitor_thread)(void*)
 		old_lsn = new_lsn;
 	}
 
-	if (difftime(time(NULL), srv_last_monitor_time) > 60) {
+	if (difftime(time(NULL),
+		     my_atomic_load32_explicit(&srv_last_monitor_time,
+					       MY_MEMORY_ORDER_RELAXED))
+	    > 60) {
```

Comment:
Could you just move this branch to `srv_monitor_thread`? It doesn't seem to make any sense to update stats in a separate thread. Then you don't need atomic operations. You may also have to move this whole branch under `srv_innodb_monitor_mutex` protection, but that shouldn't be a big problem.

Reply:
What do you think if I make `srv_last_monitor_time` a local variable in `srv_monitor_thread`?

Reply:
But then how do you fix `srv_printf_innodb_monitor()`? I think it can be called by other threads. OTOH I doubt it uses `srv_last_monitor_time` correctly (probably it should actually use `last_monitor_time`?).

Anyway this is something that needs to be understood and fixed properly. If you can make it local eventually - I'm all for it.

Reply:
Ok, you`re right. I can't make `srv_last_monitor_time` local easily. Let it be a subject of another PR.

I don't see TSAN reports now. But while this variable is accessed from different thread I guess I will with some different test in future.

Reply:
That's why I told:
> You may also have to move this whole branch under srv_innodb_monitor_mutex protection

Something like:
```
static void srv_refresh_innodb_monitor_stats(void)
{
        mutex_enter(&srv_innodb_monitor_mutex);
        if (difftime(time(NULL), srv_last_monitor_time) > 60) {
                goto end;
        }
        srv_last_monitor_time = time(NULL);
        ...
end:
        mutex_exit(&srv_innodb_monitor_mutex);
```

Then all accesses except for one will be guarded by the mutex. The last one will be `srv_last_monitor_time` initialisation, which I'm not yet so sure how to fix.

Reply:
I was inattentive, sorry. This is fixed now.